pkg/controller/jobframework/setup.go

```diff
@@ -62,24 +81,14 @@ func SetupControllers(mgr ctrl.Manager, log logr.Logger, opts ...Option) error {
 			if err != nil {
 				return fmt.Errorf("%s: %w: %w", fwkNamePrefix, errFailedMappingResource, err)
 			}
-			if _, err = mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version); err != nil {
-				if !meta.IsNoMatchError(err) {
-					return fmt.Errorf("%s: %w", fwkNamePrefix, err)
-				}
-				logger.Info("No matching API in the server for job framework, skipped setup of controller and webhook")
+			if !isAPIAvailable(context.TODO(), mgr, rayclusterAPI) {
```

Comment:
Just curious - So if the API is not available, is it by requirement that we don't proceed with starting the controller, or do we want to start the controller with informers on existing GVRs any way?

Asking this because, if we want to do the latter: instead of setting up watches and waiting for APIs to become available by blocking the manager to start, we could use [discovery client](https://github.com/kubernetes/client-go/blob/ab86e03da476bd58948f318b4f73d67d0801fcbd/discovery/discovery_client.go#L739) at the time of setup, check for API availability, and set up watches accordingly. 

Setting up informers prior or later in the process depends on whether we want the CRDs to be installed as a requirement or make it optional.